src/relayclient/DevRelayClient.ts

```diff
@@ -0,0 +1,254 @@
+import RelayClient, { RelayingResult } from './RelayClient'
+import RelayServer from '../relayserver/RelayServer'
+import HttpServer from '../relayserver/HttpServer'
+import { sleep } from '../common/utils'
+import { HttpProvider, provider } from 'web3-core'
+import GsnTransactionDetails from './types/GsnTransactionDetails'
+import KnownRelaysManager from './KnownRelaysManager'
+import RelayRegisteredEventInfo from './types/RelayRegisteredEventInfo'
+import { GSNConfig, GSNDependencies } from './GSNConfigurator'
+import { Address } from './types/Aliases'
+import { IStakeManagerInstance } from '../../types/truffle-contracts'
+import { ether } from '@openzeppelin/test-helpers'
+import stakeManagerAbi from '../common/interfaces/IStakeManager'
+import { TxStoreManager } from '../relayserver/TxStoreManager'
+import axios from 'axios'
+
+import KeyManager = require('../relayserver/KeyManager')
+
+const dayInSec = 24 * 60 * 60
+const weekInSec = dayInSec * 7
+
+import TruffleContract = require('@truffle/contract')
+import Contract = Truffle.Contract
+
+// wait for relay until waitFunction return true value, or until a timeout.
+async function waitForRelay (url: string, timeout: number, waitFunction: (res: any) => RelayRegisteredEventInfo | undefined): Promise<RelayRegisteredEventInfo> {
+  const timeoutTime = Date.now() + timeout
+  while (Date.now() <= timeoutTime) {
+    let res
+    try {
+      res = await axios.get(url)
+    } catch (e) {
+      res = e
+    }
+    const ret = waitFunction(res)
+    if (ret != null) {
+      return ret
+    }
+    await sleep(400)
+  }
+  throw new Error('waitForRelay: timed-out')
+}
+
+interface RunServerReturn {
+  httpServer: HttpServer
+  relayServer: RelayServer
+}
+
+export function runServer (
+  web3provider: provider,
+  relayHub: string,
+  devConfig: DevGSNConfig
+): RunServerReturn {
+  const relayUrl = devConfig.relayUrl ?? `http://localhost:${devConfig.relayListenPort}`
+
+  // TODO: read key-pair from temp file?
+  // (otherwise, we deploy a new relay each time)
+  const keyManager = new KeyManager({ ecdsaKeyPair: KeyManager.newKeypair(), workdir: undefined })
+  const txStoreManager = new TxStoreManager({ inMemory: true })
+
+  // @ts-ignore
+  const relayServer = new RelayServer({
+    web3provider,
+    txStoreManager,
+    keyManager,
+    // owner: relayOwner,
+    hubAddress: relayHub,
+    url: relayUrl,
+    baseRelayFee: devConfig.baseRelayFee,
+    pctRelayFee: devConfig.pctRelayFee,
+    gasPriceFactor: devConfig.gasPriceFactor,
+    devMode: devConfig.devMode
+  })
+  relayServer.on('error', (e: any) => {
+    // console.error('ERR:', e.message)
```

Comment:
Do emit if verbose, please. This commented-out outputs do shoot you in the back sometimes.